### DISCUSSION TITLE
hint: Waiting for your editor to close the file... "D:/Pliki programów na D/GitE xtensions.exe" fileeditor: D:/Pliki programów na D/GitExtensions.exe: No such file or directory error: There was a problem with the editor '"D:/Pliki programów na D/GitExtensions.exe" fileeditor'.

### DISCUSSION THREAD

**Discussion by KanapkaBezMasla** [gitextensions/gitextensions#9850] [Q&A] [answered]
### Environment

- GitExtensions version:
- GIT version: 
- OS version: 
- .NET version: 

### Issue description

When I want to make a rebase in git bash I recive an error:
"hint: Waiting for your editor to close the file... "D:/Pliki programów na D/GitExtensions.exe" fileeditor: D:/Pliki programów na D/GitExtensions.exe: No such file or directory
error: There was a problem with the editor '"D:/Pliki programów na D/GitExtensions.exe" fileeditor'."

I have already restarted gitExtensions, turned it off, and even reinstalled it to the default localisation for programs on disc C and all of it ended with the same result that I wrote above.

Thank you in advance for anwsers.

### Steps to reproduce

I do not kwon how to reproduce this situation, I recive such error always when I write a commend to bash connected with rebase, such a "git rebase -i HEAD~8"

**Comment by gerhardol** [comment, marked as answer]
Check the settings for the Git editor, can be done from GE.
It seems like the path entered is invalid. (You can use any editor that Git supports.)

![image](https://user-images.githubusercontent.com/6248932/151227793-0fb46719-7047-4e55-a03b-b929fe5c838c.png)

**Reply by vivekjoshi5** [reply]
**LATEST UPDATE: Problem is resolved by myself 🎉**
I realized it was happening because the atom program file that cmder emulator was looking for was in old directory which didn't exist as I reset my system few days ago .
Solution: I went to advanced system settings > Environment Variables> Select PATH under System Variables> Edit> New> Added the original atom directory path to  the list****

****This directory path can be known by going to File explorer> View tab> Options> Change folder & search options> View> Click radio button show hidden files, folders and drives. After this go to C:\users\<username>\AppData\Local\<App Folder>.
Copy the folder path of the editor(In my case it was Atom)
This process is for Windows 10

*Leaving it here so if anyone stumbles upon the same issue in the future(through Google Search) can fix it by using this solution. It took me almost 2 hours to resolve this. Hope your time is saved 😊*
------------------------------------------------------------------------------------------------------------------------------------------
**Original Issue**
Where is this Git Editor located ??
I am currently on Git version 2.36.1 and using Atom for writing code. I am using commander(https://cmder.net/) emulator for running git commands. I am getting the same error as that person:
 1. when commiting changes after using **git revert** &
 2. when commiting changes after resolving merging conflicts.

I couldn't locate this git editor ( that screenshot that you have shared here). Kindly help.

UPDATE:
This is what it shows but it's indeed an error because my Atom app is in folder named as **1.60.0** but the below code shows cmdr is trying to look in **folder 1.58.0** which does not exist. 

`hint: Waiting for your editor to close the file... "C:\Users\vivek\AppData\Local\atom\app-1.58.0\atom.exe" --wait: C:\Users\vivek\AppData\Local\atom\app-1.58.0\atom.exe: No such file or directory`

**Reply by gerhardol** [reply]
Settings -> Git -> Config to set the editor,
or `git config --global core.editor <tool-path>`, see https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration

**Reply by paarayac** [reply]
Setting the Editor that [gerhardol](https://github.com/gerhardol) has in the picture worked for me
